proposed/2023/nuget-package-search.md

```diff
@@ -0,0 +1,72 @@
+# NuGet Package Search
+<!-- Replace `Title` with an appropriate title for your design -->
+
+- [Nigusu](https://github.com/Nigusu-Allehu) <!-- GitHub username link -->
+- [GitHub Issue](https://github.com/NuGet/Home/issues/6060) <!-- GitHub Issue link -->
+
+## Summary
+
+<!-- One-paragraph description of the proposal. -->
+This specification will discuss three key improvements to the NuGet package search functionality. These improvements are: adding an `-ExactMatch` option to the `search` command, deprecating the `list` command in favor of `search`, and introducing a `dotnet nuget search` command to the Dotnet CLI.
+## Motivation 
+
+<!-- Why are we doing this? What pain points does this solve? What is the expected outcome? -->
+Improving the nuget search functionality across all platforms is highly requested by customers. As shown in this [issue](https://github.com/NuGet/Home/issues/5138), customers would like to have `-ExactMatch` option added to the search command. In this other [issue](https://github.com/NuGet/Home/issues/6060), we can see that it has been repeatedly requested by customers for the Dotnet CLI to adopt a search functionality. Lastly, as discussed in this [issue](https://github.com/NuGet/Home/issues/7912) the `nuget list` command is redundant with `nuget search` as both commands basically display a list of packages from a source. The only difference between the two is that the search command allows for filters. In addition, the `nuget list` command can be confusing as there is a `nuget list` command in the Dotnet CLI with a different functionality: lists configured sources and client certificates. Addressing these issues will provide users with a not confusing search functionality on both the Dotnet CLI and nuget.exe.
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+#### 1. Adding `-ExactMatch` Option: 
+A new flag, `-ExactMatch`, will be added to the `search` command. When this flag is used, it will only return packages whose names exactly match the query.
+- **Version not specified** : In this case, the latest version of the package specified is listed .\
```

Comment:
Are we sure about defaulting to latest version with `ExactMatch`? I wouldn't intuitively think that is what would happen by specifying `-ExactMatch` without a parameter. 

Our existing `list` behavior breakdown is:
1. `(no version options specified)`: list latest version
2. `-AllVersions`: list all versions
3. Exact isn't possible

So why not introduce number 3 above as `-ExactMatch <version required>` ?

Reply:
I see, it could be confusing to users. However, the current implementation of `list` does not achieve an exact match. For example, if I were to do `nuget list packageA` and a package named `PackageA`, and `PackageAmazing`  exist, both packages will be printed out. However, the user wants the exact match only. With `nuget search PackageA -ExactMatch` only the latest version of `PackageA` will be printed out as the user wants the exact match only. if they specify the version, that specific version will be printed out., or if they do `-AllVersions`, all of them will be displayed. If we were to include `-ExactVersion` as the third behaviour, users would not be able to achieve: 
1. Exact matching a package name without a version number
2. Listing all versions of an exact match, not just all matching packages.

Reply:
`-AllVersions` doesn't really work and hasn't in a while (server bugs)

IMO we should start with a more minimal version of the command.